pkg/operator/controllers/dnsmasq/scripts/99-dnsmasq-restart.gotmpl

```diff
@@ -27,12 +27,29 @@ log_dns_files() {
     log "$(echo -n "/etc/resolv.conf.dnsmasq file metadata: ") $(ls -lZ /etc/resolv.conf.dnsmasq)"
 }
 
+sanity_check() {
+    if [[ ! -f /etc/resolv.conf ]]; then
+        log "/etc/resolv.conf is missing. Sanity check failed."
+        exit 1
+    elif [[ ! -f /etc/resolv.conf.dnsmasq ]]; then
+        log "/etc/resolv.conf.dnsmasq is missing. Sanity check failed."
+        exit 1
+    fi
+}
```

Comment:
It might be worth leaving a comment in the code on why this is necessary. If the files are missing, what would be next steps to resolve that?

Reply:
`/etc/resolv.conf.dnsmasq` should only be missing on the bootstrapper. Ideally I'd like to disable this portion of the controller from being applied to the bootstrapper, and this in theory shouldn't happen.

My thinking here is to just exit early incase of the unexpected, rather than letting the script execute and log file not found errors.

On a worker or master node, if `/etc/resolv.conf.dnsmasq` doesn't exist, we probably want to restart dnsmasq anyways? So that the systemd unit `ExecStartPre` executes `aro-dnsmasq-pre.sh`.

I'd like others thoughts on this as well?

Reply:
+1

Reply:
Another case where /etc/resolv.conf.dnsmasq is missing is on node boot. eth0 comes up before dnsmasq.service, so the script gets called before dnsmasq has run for the first time.

Reply:
@ventifus I can update to check if this is the first node boot, then exit without here without performing anything and logging that information. I can also add a check to see if `dnsmasq` has ever been started and log the results appropriately.  

If `/etc/resolv.conf.dnsmasq` doesn't exist **and** `dnsmasq` hasn't been started, this is expected. If `dnsmasq` has been started, and `/etc/resolv.conf.dnsmasq` is missing, we have a problem that's unaccounted for.